lib/rst/interp_float/segmen2d_parallel.c

```diff
@@ -292,88 +293,123 @@ int IL_interp_segments_2d_parallel(
+                                     indx[tid], b[tid]);
                         }
-                    }
-                    else if (segtest == 1) {
-                        if (/* params */
-                            IL_matrix_create_alloc(
-                                params, data_local[tid]->points,
-                                data_local[tid]->n_points - 1, matrix[tid],
-                                indx[tid], A[tid]) < 0) {
-                            some_thread_failed = -1;
+                        else {
                             continue;
                         }
+                    } /* cv: skip one point */
+
+                    if (params->create_devi) {
+                        target_point.x = data_local[tid]->points[point_index].x;
+                        target_point.y = data_local[tid]->points[point_index].y;
+                        target_point.z = data_local[tid]->points[point_index].z;
                     }
-                    if (!params->cv) {
-                        for (i = 0; i < data_local[tid]->n_points; i++) {
-                            b[tid][i + 1] = data_local[tid]->points[i].z;
-                        }
-                        b[tid][0] = 0.;
-                        G_lubksb(matrix[tid], data_local[tid]->n_points + 1,
-                                 indx[tid], b[tid]);
-                        /* put here condition to skip error if not needed */
-                        params->check_points(params, data_local[tid], b[tid],
-                                             ertot, zmin, dnorm, skip_point);
-                    }
-                    else if (segtest == 1) {
-                        for (i = 0; i < data_local[tid]->n_points - 1; i++) {
-                            b[tid][i + 1] = data_local[tid]->points[i].z;
+
+                    /* x, y, z is required input, while xmm, ymm, err output*/
+                    pointz = target_point.z;
+                    params->check_points(params, data_local[tid], b[tid], ertot,
+                                         zmin, dnorm, &target_point);
+
+                    err = target_point.z;
+                    target_point.z = pointz + zmin;
+                    xmm = target_point.x;
+                    ymm = target_point.y;
+
+                    /* write out vector (point), if the point is inside the
+                     * region*/
+                    if (xmm >= data_local[tid]->x_orig + params->x_orig &&
+                        xmm <= data_local[tid]->xmax + params->x_orig &&
+                        ymm >= data_local[tid]->y_orig + params->y_orig &&
+                        ymm <= data_local[tid]->ymax + params->y_orig) {
+                        /* vect append, count, vect_write, db_execute will have
+                         * conflicts between threads */
+#pragma omp critical
+                        {
+                            params->wr_point(target_point, err);
```

Comment:
I would also check if `params->wr_point` is not `NULL` because it's set only for cv calculation.

Reply:
I removed it in the new commit.